terraform/evaluate.go

```diff
@@ -758,11 +758,12 @@ func (d *evaluationStateData) GetResource(addr addrs.Resource, rng tfdiags.Sourc
 			}
 
 			// If our provider schema contains sensitive values, mark those as sensitive
+			afterMarks := change.AfterValMarks
 			if schema.ContainsSensitive() {
-				val = markProviderSensitiveAttributes(schema, val)
+				afterMarks = append(afterMarks, getValMarks(schema, val, nil)...)
```

Comment:
Is it possible to get rid of the `markProviderSensitiveAttributes` then? (a quick read of this file suggested its only a wrapper around this `getValMarks` (which could perhaps be more descriptively named) method?)